app/controllers/casa_admins_controller.rb

```diff
@@ -0,0 +1,30 @@
+class CasaAdminsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :must_be_admin, only: [:new, :create]
+
+  def new
+    @casa_admin = CasaAdmin.new
+  end
+
+  def create
+    @casa_admin = CasaAdmin.new(casa_admin_params.merge(casa_admin_values) )
+
+    if @casa_admin.save
+      @casa_admin.invite!
+      flash[:notice] = "New Admin created."
+      redirect_to root_path
+    else
+      render new_casa_admin_path
+    end
+  end
+
+  private
+
+  def casa_admin_values
+    {password: SecureRandom.hex(10), casa_org_id: current_user.casa_org_id}
```

Comment:
See above, `casa_org_id: current_user.casa_org_id` could be simplified to `casa_org: current_organization` if the `before_action :require_organization!` is implemented.